### ISSUE TITLE
Not shutting down correctly on Windows

### ISSUE THREAD

**Issue by bradharms** [electron-userland/electron-webpack#87]
I have a project that is (mostly) based on the quick start repo, but when I run it on Windows and close the main window, my command prompt leaves me with a stylized "Terminate batch job (Y/N)?" which keeps the process running. Eventually it will shut down on its own. This problem was not present when I was using Electron by itself.

This is my `src/main/index.js`:

```javascript
// @ts-check

import { app, BrowserWindow } from 'electron'
import url from 'url'
import path from 'path'

const isDevelopment = process.env.NODE_ENV !== 'production'

let mainWindow

function createMainWindow() {
  mainWindow = new BrowserWindow({
    width: 800,
    height: 550,
    autoHideMenuBar: true,
  })
  
  const url_ = isDevelopment ?
    `http://localhost:${process.env.ELECTRON_WEBPACK_WDS_PORT}` :
    url.format({
      pathname: path.join(__dirname, 'index.html'),
      protocol: 'file:',
      slashes: true
    })
  
  if (isDevelopment) {
    mainWindow.webContents.openDevTools({ mode: "detach" })
  }
  
  mainWindow.loadURL(url_)

  mainWindow.on('closed', () => {
    mainWindow = null
  })

  mainWindow.webContents.on('devtools-opened', () => {
    mainWindow.focus()
    setImmediate(() => {
      mainWindow.focus()
    })
  })
}

app.on('window-all-closed', () => {
  if (process.platform !== 'darwin') {
    app.quit()
  }
})

app.on('activate', () => {
  if (mainWindow === null) {
    createMainWindow()
  }
})

app.on('ready', createMainWindow)
```

**Comment by loopmode** [comment]
Yes I have noticed this too. I got used to it as I have a good workaround: I just kill all node processes..

Basically, it's the wmic command: `wmic process where name like "node" delete` (see https://www.cs.cmu.edu/~tgp/scsadmins/winadmin/WMIC_Queries.txt)

Since I use bash anyways, I keep this one in my .bashrc:
```
kill() {
    echo ">> Enter a process name and press [ENTER] (partial matching)"
    read name
    wmic process where "name like '%$name%'" delete
}
```
So.. I just launch a new console and go "kill, enter, node, enter" and it's gone right away.
On my other computer, I even have a shortcut to a batch file pinned to the start menu, that's handy too.

But, of course these are just workarounds, and the actual issue remains.

**Comment by awesson** [comment]
I see this as well. Not ideal.

If I need to use that same cmd promt right away, CTRL+C will kill it. Otherwise, like you said, if you wait a few seconds it will eventually end.

**Comment by loopmode** [comment]
I noticed that - at least for me - the issue does not occur if I launched manually, from a git bash, via `yarn dev`.
Whenever I launch differently, e.g. via shortcuts, batchfiles, from a bash script, via custom "yarn dev" folder context menu entry, via other consoles (I use ConEmu) - I do have the issue.

So for me, this "issue" is solved: No more tricky shortcuts for launching.
I just launch the git bash and run `yarn dev`, then after `Ctrl+C` everything quits right away without delay.

**Comment by praschl** [comment]
I am experiencing that, too. After working a while, i see my computers performance degrade, and when investigating, I see some electron processes running.

I am working on Windows 10 with VS Code.
